extra/FindNanopb.cmake

```diff
@@ -360,6 +391,10 @@ foreach(FIL ${_nanopb_hdrs})
   list(APPEND NANOPB_HDRS "${${FIL}__nano_pb_file}")
 endforeach()
 
+# Create the library target
+add_library(nanopb STATIC ${NANOPB_SRCS})
+target_include_directories(nanopb PUBLIC ${NANOPB_INCLUDE_DIRS})
+
```

Comment:
You might want to add
```
target_compile_features(nanopb PUBLIC c_std_11)
```
here as by default `_Static_Assert ` is used which warns depending on compiler:
```
nanopb/pb.h:186:42: warning: ISO C99 does not support '_Static_assert' [-Wpedantic]
  186 | #      define PB_STATIC_ASSERT(COND,MSG) _Static_assert(COND,#MSG);
```